staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/grpc_service.go

```diff
@@ -53,13 +51,8 @@ type gRPCService struct {
 }
 
 // NewGRPCService returns an envelope.Service which use gRPC to communicate the remote KMS provider.
-func NewGRPCService(endpoint string, callTimeout time.Duration) (Service, error) {
```

Comment:
this changes the behavior of this method without changing its signature. That's worth a release note at the very least. How does this behave if someone keeps calling it with the previous parameter value? Does it error in ways that make it obvious what the caller needs to adjust?

I think it's also worth exporting the parse method so callers can call the separate parse method and retain equivalent behavior.

Reply:
What do you think about changing the signature to:
func NewGRPRCService(connection *grpc.ClientConn) Service, and moving the responsibility to construct ClientConn to the config package.
This will allow us to:
1. Abstracts KMS Service from the implementation detail that (today) we only support Unix Domain Socket as a mechanism to communicate with plugins. 
2. Construct different connections based on different use-cases. For example, connections for healthz checks should have FailFast set to true, while the main connection to false.

Reply:
that seems reasonable as well